packages/shared/src/features/onboarding/shared/FunnelStepper.tsx

```diff
@@ -185,6 +185,11 @@ export const FunnelStepper = ({
     funnel?.parameters?.banner?.stepsToDisplay,
   ]);
 
+  const steps = useMemo(
+    () => funnel?.chapters?.flatMap((chapter) => chapter?.steps),
+    [funnel?.chapters],
+  );
```

Comment:
Was not necessarily part of the re-mount issue, but it is much cleaner to render the elements with a flat map (less indents and keys to manage).